packages/driver/src/cross-origin/patches/cookies.js

```diff
@@ -13,43 +56,59 @@
 // - On an interval, get the browser's cookies for the given domain, so that
 //   updates to the cookie jar (via http requests, cy.setCookie, etc) are
 //   reflected in the document.cookie value.
-export const patchDocumentCookie = (Cypress, window) => {
-  const setAutomationCookie = (toughCookie) => {
-    const { superDomain } = Cypress.Location.create(window.location.href)
-    const automationCookie = Cypress.Cookies.toughCookieToAutomationCookie(toughCookie, superDomain)
+export const handleDocumentCookie = (Cypress) => {
+  // if cookies were sent before the spec bridge was created, they'll come
+  // through as state('crossOriginCookies'), so use them then reset them
+  const cookies = Cypress.state('crossOriginCookies')
```

Comment:
is this state unique per spec bridge or could we get this state from the primary cypress instance through a post message?

Reply:
instead of passing state here, could we call out to the primary cypress instance to retrieve these cookies?

Reply:
The trick is we need them synchronously on page load. Post message won't work because it's async. I think what we'll need to do is hold on to them server-side and include them directly in the injection when we render it.

Reply:
We can do that. I already do it for modify obstructive code config.